### PR TITLE
[core] #457 jobs for indexing class types in a java file

### PR DESCRIPTION
**Pull request by PiyushMishra**

### CODE REVIEW

File: core/src/main/scala/com/kodebeagle/indexer/IndexEntities.scala [hunk 1]
```diff
@@ -40,6 +40,9 @@ case class HighLighter(lineNumber: Int, startColumn: Int, endColumn: Int)
 
 case class SourceFile(repoId: Int, fileName: String, fileContent: String)
 
+case class SourceFileWitTypes(repoId: Int, fileName: String, fileContent: String,
```
Comment (sachintyagi22) [inline core/src/main/scala/com/kodebeagle/indexer/IndexEntities.scala]: Use single case class only.

File: core/src/main/scala/com/kodebeagle/indexer/JavaASTBasedIndexerForMethods.scala [hunk 2]
```diff
@@ -37,16 +37,32 @@ class JavaASTBasedIndexerForMethods extends JavaASTBasedIndexer {
     (imports, createMethodIndexEntries(parser, tokensMap))
   }
 
+  private def extractTokensWithTypesASTParser(excludePackages: Set[String],
```
Comment (sachintyagi22) [inline core/src/main/scala/com/kodebeagle/indexer/JavaASTBasedIndexerForMethods.scala]: Modify the original method along with test cases.

File: core/src/main/scala/com/kodebeagle/spark/SparkIndexJobHelper.scala [hunk 3]
```diff
@@ -39,6 +39,15 @@ object SparkIndexJobHelper {
     map.map(x => SourceFile(repo2.id, fileNameToURL(repo2, x._1), x._2)).toSet
   }
 
+  def mapToSourceJavaFiles(repo: Option[Repository],
```
Comment (sachintyagi22) [inline core/src/main/scala/com/kodebeagle/spark/SparkIndexJobHelper.scala]: Better to use single method only for both java and scala.

**Comment by PiyushMishra** [comment]
this pull request is closed, because we will make file types as part of metadata index.

**Pull request closed without merging**